arangod/Pregel/Worker/ExecutionStates/StoredState.cpp

```diff
@@ -0,0 +1,90 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Aditya Mukhopadhyay
+////////////////////////////////////////////////////////////////////////////////
+
+#include "StoredState.h"
+#include "FatalErrorState.h"
+#include "Pregel/Worker/State.h"
+#include "ComputingState.h"
+#include "CleaningUpState.h"
+#include "Pregel/Algos/WCC/WCCValue.h"
+#include "Pregel/SenderMessage.h"
+#include "Pregel/Algos/SCC/SCCValue.h"
+#include "Pregel/Algos/HITS/HITSValue.h"
+#include "Pregel/Algos/HITSKleinberg/HITSKleinbergValue.h"
+#include "Pregel/Algos/EffectiveCloseness/ECValue.h"
+#include "Pregel/Algos/DMID/DMIDValue.h"
+#include "Pregel/Algos/LabelPropagation/LPValue.h"
+#include "Pregel/Algos/SLPA/SLPAValue.h"
+#include "Pregel/Algos/ColorPropagation/ColorPropagationValue.h"
+#include "Pregel/Algos/DMID/DMIDMessage.h"
+
+using namespace arangodb;
+using namespace arangodb::pregel;
+using namespace arangodb::pregel::worker;
+
+template<typename V, typename E, typename M>
+Stored<V, E, M>::Stored(actor::ActorPID self, WorkerState<V, E, M>& worker)
+    : self(std::move(self)), worker{worker} {}
+
+template<typename V, typename E, typename M>
+auto Stored<V, E, M>::receive(actor::ActorPID const& sender,
+                              worker::message::WorkerMessages const& message,
+                              Dispatcher dispatcher)
+    -> std::unique_ptr<ExecutionState> {
+  if (std::holds_alternative<worker::message::Cleanup>(message)) {
```

Comment:
yes, you can receive a Cleanup message here, but you should actually be able to receive a Cleanup message in every execution state and do the cleanup, because a pregel run can be canceled at any time by the user.

Reply:
Fixed in 41b3ac94a31112e68185e219bf6af0c9e10ffd4c